src/mapgen/mapgen.cpp

```diff
@@ -1021,10 +1047,17 @@ void GenerateNotifier::getEvents(
 	}
 }
 
+void GenerateNotifier::getUD(
+	std::map<std::string, std::string> &map) const
+{
+	map = m_notify_ud;
```

Comment:
If this is supposed to avoid threading issues, there should be a note to state that. Otherwise you could return a const reference or make this function to "pop" the value into `map` using `std::move`.

Reply:
Didn't think about that at all, will change to const-ref.